openmc/trigger.py

```diff
@@ -0,0 +1,98 @@
+from numbers import Real
+from xml.etree import ElementTree as ET
+import sys
+import warnings
+from collections.abc import Iterable
+
+import openmc.checkvalue as cv
+
+
+class Trigger(object):
+    """A criterion for when to finish a simulation based on tally uncertainties.
+
+    Parameters
+    ----------
+    trigger_type : {'variance', 'std_dev', 'rel_err'}
+        Determine whether to trigger on the variance, standard deviation, or
+        relative error of scores.
+    threshold : float
+        The threshold for the trigger type.
+
+    Attributes
+    ----------
+    trigger_type : {'variance', 'std_dev', 'rel_err'}
+        Determine whether to trigger on the variance, standard deviation, or
+        relative error of scores.
+    threshold : float
+        The threshold for the trigger type.
+    scores : list of str
+        Scores which should be checked against the trigger
+
+    """
+
+    def __init__(self, trigger_type, threshold):
+        self.trigger_type = trigger_type
+        self.threshold = threshold
+        self._scores = []
+
+    def __eq__(self, other):
+        return str(self) == str(other)
```

Comment:
Generally not a good idea to implement `__eq__` like this. The default `__eq__` method will at least check the properties on a deeper level and not just cast them to strings.